docs/design/connection-tester.md

```diff
@@ -0,0 +1,50 @@
+# Connection Tester
+
+The goal of this document is to sketch out possible approaches for implementing _Connection testing_ in the context of iPaaS. This backend concept is introduced to support the use case to quickly test whether a given `Connection`  (which is the combination of a `Connector` and a `ConnectorConfiguration`) is actually usable. The typical use case is, that the user pushes a button "Test connection" on the "Connection Details" page which actually triggers the test of the connection and that can be either successful or failing. When it fails, a detailed error message describes not only the error itself but also how to resolve the error (e.g. "Cannot connect because of invalid credentials"). 
+
+## Communication flow
+
+When the user triggers this test function from within the UI, a backend call to `ipaas-rest` will be issued, which in turn will trigger a (still to be defined) method on the referenced connector with the concrete configuration given. This method, let's call it `ping()`, must be side effect free, which should be possible for most endpoints. The method takes the configuration as argument for the connector and returns a result objects which contains the result of the check (ok / nok) and in case of a failure the error message (question: Should we already think about I18N here ?)
```

Comment:
@KurtStam I see the use case, but I would keep that separate for the moment, mostly also because you can't really define an API which is uniform over all connectors that takes a payload.

I think this use case is better covered when you test a whole integration. Testing the connector alone _functionally_ (i.e. doing what it is supposed) shouldn't be the task of the user but of us. 

So I think we should distinguish between:

* Testing whether a given configuration can connect with the backend. When testing a connection the user should be safe that there no changes will happen so that he can press the button as often as he wants.
* Testing an integration with a test message so that the user can manually verify the result. Sending a test message will always have a side effect, so the user knows this when pushing that button (which btw. should appear somewhere else in UI as it is a different use case),